### DISCUSSION TITLE
Feature ideas

### DISCUSSION THREAD

**Discussion by nahar-emet** [Budibase/budibase#4252] [Ideas]
this is a list of features that I wished where there when working on my present project: developing a system built around budibase with a postgres db to manage orders, manufacturing and distribution for a small startup.

a switch / if else and a merge block in automations would make them a lot more flexible and much more powerful. I would recommend looking at Houdini and Nuke fx from the VFX industry for inspirations for the automations tab. They have been doing node based software for the past 20 years. 

ability to add more color to the ui to make things more visually appealing ex:
       small colorable rounded elements that can have 1 piece of information  akin mui chips so as to be able to highlight information.
       conditional formatting for text elements and the ability to use badges in tables similar to air tables. 
      add colors to cards. 
      costume colors for charts.

ability to configure data elements in charts: ex a donut chart with numbers instead of percentages.

a camera widget and QR reader widget: im sure that both would be tricky to implement but would be amazing to have. 

thank you for your amazing work i have tried all of the other solutions and budibase is by far the best one from a user  and design perspective and im sure that will have a bright future.

**Comment by tongclement** [comment]
+1 for the QR Reader
